src/cpp/fastdds/xtypes/type_conversion/dyn_type_tree.hpp

```diff
@@ -0,0 +1,252 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file dyn_type_tree.hpp
+ */
+
+#ifndef _FASTDDS_DDS_XTYPES_DYN_TYPE_TREE_HPP_
```

Comment:
It should be consistent since #4958, at least for the public headers (the ones under `include`). We ought to follow that convention for all new headers (private or otherwise).